### PR TITLE
common: change language field to IETF BCP 47 language tag

### PR DESCRIPTION
**Pull request by prusnak**
Let's use IETF BCP 47 language tag as language field.

This allows us to use different dialects (such as `pt-PT`, `pt-BR`) of the same language. Also, the tag is shorter than the whole name and its usage is more standardized (i.e. browser's `navigator.language` returns the value(s) in this format).

### CODE REVIEW

**Comment by prusnak** [comment]
CI failure in Travis is bogus and fails on master too.

**Comment by tsusanka** [comment]
Travis should be fine after rebase.

File: legacy/firmware/config.c [hunk 1]
```diff
@@ -665,7 +665,17 @@ bool config_getLabel(char *dest, uint16_t dest_size) {
...
+    if (dest_size == 7 && (strcmp(dest, "english") != 0)) {
+      // fallthrough -> return "en-US"
+    } else {
+      // other language -> return the value
+      return true;
+    }
+  }
+  strcpy(dest, "en-US");
+  dest_size = 5;
+  return true;
```
Comment (tsusanka) [inline legacy/firmware/config.c]: If `config_get_string` returns false, we still return `en-US`? Is it desired? It makes sense as a default value

Comment (prusnak) [reply legacy/firmware/config.c]: Yes, I want `en-US` to be the default even if no value is set. (We have the same behaviour in TT - we discard any values to be set and return `en-US` always).

**Pull request merged**